Source/WebKit/UIProcess/Automation/WebAutomationSession.h

```diff
@@ -171,6 +172,13 @@ friend class WebDriverBidiProcessor;
     void navigationFailedForFrame(const WebFrameProxy&, std::optional<WebCore::NavigationIdentifier>);
     void navigationAbortedForFrame(const WebFrameProxy&, std::optional<WebCore::NavigationIdentifier>);
     void fragmentNavigatedForFrame(const WebFrameProxy&, std::optional<WebCore::NavigationIdentifier>);
+    void contextCreatedForPage(const WebPageProxy&);
+    void contextCreatedForFrame(const WebFrameProxy&);
+    void contextDestroyedForPage(const WebPageProxy&);
+    void contextDestroyedForFrame(const WebFrameProxy&);
+    void recursivelyEmitContextCreated(const FrameTreeNodeData&, std::optional<String>&& parentContext);
+    std::pair<String, String> getClientWindowAndUserContext(const WebPageProxy&);
+    RefPtr<WebPageProxy> getOpenerPage(const WebPageProxy&);
```

Comment:
These last three don't need to be in the header file, they can be private to the .cpp file.